android/modules/ui/src/java/ti/modules/titanium/ui/android/DrawerLayoutProxy.java

```diff
@@ -24,170 +26,264 @@
-    }
-
-    @Kroll.method
-    @Kroll.getProperty
-    public boolean getIsRightVisible() {
-        return drawer != null && drawer.isRightVisible();
-    }
-
-    @Kroll.method
-    @Kroll.setProperty
-    public void setLeftWidth(Object arg) {
-        setPropertyAndFire(TiC.PROPERTY_LEFT_WIDTH, arg);
-    }
-
-    @Kroll.method
-    @Kroll.setProperty
-    public void setLeftView(Object arg) {
-        setPropertyAndFire(TiC.PROPERTY_LEFT_VIEW, arg);
-    }
-
-    @Kroll.method
-    @Kroll.setProperty
-    public void setRightWidth(Object arg) {
-        setPropertyAndFire(TiC.PROPERTY_RIGHT_WIDTH, arg);
-    }
-
-    @Kroll.method
-    @Kroll.setProperty
-    public void setRightView(Object arg) {
-        setPropertyAndFire(TiC.PROPERTY_RIGHT_VIEW, arg);
-    }
-
-    @Kroll.method
-    @Kroll.setProperty
-    public void setCenterView(Object arg) {
-        setPropertyAndFire(TiC.PROPERTY_CENTER_VIEW, arg);
-    }
-
-    @Kroll.method
-    @Kroll.getProperty
-    public boolean getDrawerIndicatorEnabled() {
-        if (hasProperty(TiC.PROPERTY_DRAWER_INDICATOR_ENABLED)) {
-            return (Boolean) getProperty(TiC.PROPERTY_DRAWER_INDICATOR_ENABLED);
-        }
-        return true;
-    }
-
-    @Kroll.method
-    @Kroll.setProperty
-    public void setDrawerIndicatorEnabled(Object arg) {
-        setPropertyAndFire(TiC.PROPERTY_DRAWER_INDICATOR_ENABLED, arg);
-    }
-
-    @Kroll.method
-    @Kroll.getProperty
-    public int getDrawerLockMode() {
-        if (hasProperty(TiC.PROPERTY_DRAWER_LOCK_MODE)) {
-            return (Integer) getProperty(TiC.PROPERTY_DRAWER_LOCK_MODE);
-        }
-        return LOCK_MODE_UNDEFINED;
-    }
-
-    @Kroll.method
-    @Kroll.setProperty
-    public void setDrawerLockMode(Object arg) {
-        setPropertyAndFire(TiC.PROPERTY_DRAWER_LOCK_MODE, arg);
-    }
-
-    @Kroll.method
-    public void interceptTouchEvent (TiViewProxy view, Boolean disallowIntercept){
-        view.getOrCreateView().getOuterView().getParent().requestDisallowInterceptTouchEvent(disallowIntercept);
-    }
-
-    @Kroll.method
-    @Kroll.getProperty
-    public boolean getToolbarEnabled() {
-        if (hasProperty(TiC.PROPERTY_TOOLBAR_ENABLED)) {
-            return (Boolean) getProperty(TiC.PROPERTY_TOOLBAR_ENABLED);
-        }
-        return true;
-    }
-
-    @Kroll.method
-    @Kroll.setProperty
-    public void setToolbarEnabled(Object arg) {
-        setPropertyAndFire(TiC.PROPERTY_TOOLBAR_ENABLED, arg);
-    }
+	@Kroll.constant public static final int LOCK_MODE_LOCKED_CLOSED = DrawerLayout.LOCK_MODE_LOCKED_CLOSED;
+	@Kroll.constant public static final int LOCK_MODE_LOCKED_OPEN = DrawerLayout.LOCK_MODE_LOCKED_OPEN;
+	@Kroll.constant public static final int LOCK_MODE_UNLOCKED = DrawerLayout.LOCK_MODE_UNLOCKED;
+	@Kroll.constant public static final int LOCK_MODE_UNDEFINED = DrawerLayout.LOCK_MODE_UNDEFINED;
+
+	private static final int MSG_FIRST_ID = KrollProxy.MSG_LAST_ID + 1;
```

Comment:
There is a potential future bug here. The `MSG_FIRST_ID` constant should be using `TiViewProxy.MSG_LAST_ID` and not `KrollProxy.MSG_LAST_ID` in order to avoid ID collision.

```java
private static final int MSG_FIRST_ID = TiViewProxy.MSG_LAST_ID + 1;
```